src/semafour.cc

```diff
@@ -103,17 +103,18 @@ int Semafour::TryWait() {
     r = sem_trywait(_sem);
   } while (r == -1 && errno == EINTR);
 
-  bool locked = r == -1 && errno == EAGAIN;
+  // No error!
+  if (r == 0) {
```

Comment:
Doing early returns like this is a habit that I've taken. It's not as useful as in bigger examples though - let me know you'd rather use plain ol' `else`.

Also, let me know if I need to remove the blank lines.

Reply:
Early returns in general are fine 😄